### ISSUE TITLE
Propagate limit_amplitude keyword argument to parametric pulses

### ISSUE THREAD

**Issue by taalexander** [Qiskit/qiskit-terra#6544]
### What is the expected enhancement?
As described [here](https://github.com/Qiskit/qiskit-terra/pull/6345#pullrequestreview-666847848) the `limit_amplitude` keyword argument should be propagated to the constructor of existing parametric pulses. 

For example,

```python
from qiskit.pulse import Gaussian
amp = 10.0
Gaussian(100, amp, 1.0, limit_amplitude=False)
```

**Comment by jwoehr** [comment]
`class Pulse` defaults `limit_amplitude` property to `True`.

Should that `True` value be the propagated as the default to `class ParametricPulse` and its specializations, or should it be as in this Issue's header, a default of `False`?

**Comment by taalexander** [comment]
The default value should be `True`. The example in the issue was intended to demonstrate how to disable amplitude checking.

**Comment by dhruvbhq** [comment]
Hi @taalexander, @jwoehr. May I work on this?

**Comment by jwoehr** [comment]
It's open source, go ahead, @dhruvbhq

**Comment by dhruvbhq** [comment]
Hi, I have raised a PR, but I have a query. 

In the method validate_parameters of class Drag (derived from ParametricPulse), there is an assertion/check "Beta is too large: pulse amplitude norm exceeds 1". It is checking whether the value of 'Beta' causes the maximum amplitude to exceed 1. For now, I have bypassed this check in case 'limit_amplitude' is false, as in this case it doesn't make sense (to me) to enforce this check. However, I think this would leave 'Beta' completely unconstrained. 

In case limit_amplitude is false, is there a need to put any additional check on the value of 'Beta'?
